### PR TITLE
Fix excon errors namespace for excon >= 0.50

### PR DESCRIPTION
**Pull request by lebedev-yury**
Excon changes errors namespace from Excon::Errors to Excon::Error since
version 0.50.x

### CODE REVIEW

**Comment by lebedev-yury** [comment]
https://github.com/excon/excon/commit/e300458f2d9330cb265baeb8973120d08c665d9c

File: lib/docker/connection.rb [hunk 1]
```diff
@@ -38,17 +38,17 @@ def request(*args, &block)
...
+  rescue Excon::Error::Unauthorized => ex
     raise UnauthorizedError, ex.response.body
-  rescue Excon::Errors::NotFound => ex
+  rescue Excon::Error::NotFound => ex
     raise NotFoundError, ex.response.body
-  rescue Excon::Errors::Conflict => ex
+  rescue Excon::Error::Conflict => ex
     raise ConflictError, ex.response.body
-  rescue Excon::Errors::InternalServerError => ex
+  rescue Excon::Error::InternalServerError => ex
```
Comment (dalehamel) [inline lib/docker/connection.rb]: This should be Excon::Errors::InternalServerErrorError (yep, believe it or not)

https://github.com/excon/excon/commit/13758ec567bc090f4fba6e433f191fb92434637c

**Comment by dalehamel** [comment]
Tests look to be failing, but :+1: to getting this change out

**Comment by lebedev-yury** [comment]
@dalehamel I fixed the wrong name for `Excon::Errors::InternalServerErrorError`

It's actually strange that the build is failing, look like the docker credentials are not exported correctly for this build by travis

File: lib/docker/connection.rb (hunk 1 above)
Comment (lebedev-yury) [inline lib/docker/connection.rb]: I'm actually not sure about this:

https://github.com/excon/excon/commit/d2a550c052b0e904d247d85314907bf366213c8e

**Comment by lebedev-yury** [comment]
Yes, the error should actually be `Excon::Error::InternalServerError`:

```
irb(main):001:0> require 'docker-api'
=> true
irb(main):002:0> Excon::Error::InternalServerError
=> Excon::Error::InternalServerError
irb(main):003:0> Excon::Error::InternalServerErrorError
NameError: uninitialized constant Excon::Error::InternalServerErrorError
```

**Comment by dalehamel** [comment]
The plural form needs error twice

**Comment by lebedev-yury** [comment]
Yes, you are right, but we could use the single form here:

```
irb(main):002:0> Excon::Errors::InternalServerErrorError
=> Excon::Error::InternalServerError
```

**Comment by lebedev-yury** [comment]
@dalehamel so what about this PR? The build is failing since the credentials are not set for private docker repos. Should I close it, or it could be merged?

**Comment by tlunter** [comment]
@lebedev-yury is this still an issue past excon v0.51.0?

**Comment by lebedev-yury** [comment]
@tlunter yes, excon still uses `Error` module for errors instead of `Errors` as before

**Comment by tlunter** [comment]
@lebedev-yury Given the backwards compatibility layer added by Excon, we should be fine using `Errors`, right? I'd rather not force everyone to upgrade excon if such backwards compatibility exists.

**Comment by lebedev-yury** [comment]
@tlunter yep, totally agree

**Pull request closed without merging**